actions/geoip.py

```diff
@@ -0,0 +1,112 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import ipaddress
+import geoip2.database
+
+from st2actions.runners.pythonrunner import Action
+
+
+class GeoIpAction(Action):
+    def _get_databases(self):
+        """
+        Try to open all the GeoIP2 databases we need.
+        """
+
+        try:
+            reader_isp = geoip2.database.Reader(self.config['isp_db'])
+        except IOError:
+            reader_isp = None
+
+        try:
+            reader_city = geoip2.database.Reader(self.config['city_db'])
+        except IOError:
+            reader_city = None
+
+        return (reader_isp, reader_city)
+
+    def run(self, ip_addresses):
+        """
+        Return GeoIP information about an IP address
+
+        Args:
+        - ip_address: The IP address to validate.
+
+        Raises:
+        - ValueError: On invalid database.
+
+        Returns:
+        dict: with keys of IP address containing a dict of the
+        GeoIP information.
+        """
+
+        results = {"ok": False, "geoip": {}}
```

Comment:
New StackStrom versions support returning `(status, result)` tuple so you can utilize this instead.

But it seems like this might not actually be needed if you throw a couple of lines below :)